### PR TITLE
Feature/879 Add a creation/modification dialog for edge modules

### PR DESCRIPTION
**Pull request by audserraCGI**
## Description

What's new?

- #879 

## What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Tests
- [ ] Other

### CODE REVIEW

File: src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs [hunk 1]
```diff
@@ -30,11 +31,16 @@ public class IoTEdgeModule
         /// <summary>
         /// The module environment variables.
         /// </summary>
-        public Dictionary<string, string> EnvironmentVariables { get; } = new Dictionary<string, string>();
+        public List<IoTEdgeModuleEnvironmentVariable> EnvironmentVariables { get; set; } = new List<IoTEdgeModuleEnvironmentVariable>();
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs:34]: What is the reason to remove Dictionary here ?

File: src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs [hunk 2]
```diff
@@ -30,11 +31,16 @@ public class IoTEdgeModule
...
 
         /// <summary>
         /// The module identity twin settings.
         /// </summary>
-        public Dictionary<string, string> ModuleIdentityTwinSettings { get; } = new Dictionary<string, string>();
+        public List<IoTEdgeModuleTwinSetting> ModuleIdentityTwinSettings { get; set; } = new List<IoTEdgeModuleTwinSetting>();
```
Comment (kbeaugrand) [inline src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs:39]: Same here : What is the reason to remove Dictionary here ?

**Review by kbeaugrand — CHANGES_REQUESTED** [review]

File: src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs (hunk 1 above)
Comment (audserraCGI) [reply src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs:34]: There were issues with editing keys in the Dictionary type (and mapping them to the editing table in the dialog), so I created new models (EnvironmentVariable, TwinSettings and Commands) and used List of theses models instead.

Comment (kbeaugrand) [reply src/AzureIoTHub.Portal/Shared/Models/v1.0/IoTEdgeModule.cs:34]: ok

**Pull request merged**